File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.cs
using System;

// 문자열비교

public class Solution {
    public string solution(string code) {
        string answer = "";
        int mode = 0;
        for (int i=0; i<=code.Length-1; i++) {
            if(mode == 0) {
                if (code[i].ToString() != "1") {
                    if (i%2 == 0) {
                        answer += code[i];
                    }
                }
                else if (code[i].ToString() == "1") {
                    mode = 1;
                }
            }
            else if(mode == 1) {
                if (code[i].ToString() != "1") {
                    if (i%2 == 1) {
                        answer += code[i];
                    }
                }
                else if (code[i].ToString() == "1") {
                    mode = 0;
                }
        
            }
        }
        if (answer == "") {
            return "EMPTY";
        }
        return answer;
    }
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string code) 
{
    int mode = 0;
    string ret = "";
    for(int i=0; i<code.length(); i++)
    {
        if(mode == 0)
        {
            if(code[i] != '1')
            {
                if(i%2 == 0)
                {
                    ret += code[i];
                }
            }
            else if(code[i] == '1')
            {
                mode = 1;
            }          
        }
        else if(mode == 1)
        {
            if(code[i] != '1')
            {
                if(i%2 == 1)
                {
                    ret += code[i];
                }
            }
            else if(code[i] == '1')
            {
                mode = 0;
            }          
        }
    }
    if(ret.empty())
    {
        ret = "EMPTY";
    }
    return ret;
}